Array shift (#3184)
Codecademy/docs: content/php/concepts/arrays/terms/array_shift/array-shift.md

---
Title: 'array_shift()'

> Description: 'Shifts/pops an element off the beginning of an array.'

Reviewer: ```suggestion
Description: 'Pops an element off the beginning of an array.'
```

Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Data Types'
CatalogContent:
  - 'learn-php'
  - 'paths/computer-science'
---

> The **`array_shift()`** shifts/pops the first value of the array off and returns it, shortening the array by one element and shifting each element down. Numerical keys will be reset to start counting from 0, whereas the remaining literal keys will not be affected.

Reviewer: ```suggestion
The **`array_shift()`** pops the first value of the array off and returns it, shortening the array by one element and shifting each element down. Numerical keys will be reset to start counting from `0`, whereas the remaining literal keys will not be affected.
```

## Syntax

```pseudo

> array_shift(array &$array)

Reviewer: ```suggestion
array_shift($array)
```

```

- `$array` is the input array.

## Example

The following example shifts the first element in `$counting_rhyme` and assigns the returned value to the `$removed` variable, then it prints out `$removed` and the modified `$counting_rhyme` array.

```php
<?php
    $counting_rhyme = ["Eeny", "Meeny", "Miny", "Moe"];
    $removed = array_shift($counting_rhyme); 

>     echo $removed . "\n"; //Prints: Eeny
>     print_r($counting_rhyme) //Prints: the modified array

Reviewer: ```suggestion
    echo $removed . "\n";
    print_r($counting_rhyme)
```

Reviewer: the comments are unnecessary because the output is included below. Also update the indentations to use 2 spaces not tabs

?>
```

The above code results in the following output:

```shell
Eeny
Array
(

>     [0] => Meeny

Reviewer: update to use 2 spaces

    [1] => Miny
    [2] => Moe
)
```

## Codebyte Example

> In the code snippet below `array_shift()` is used twice. First time it takes the one element with a numerical key, second time with a literal key. After that the `$num_array` is printed out. Notice that a string of a numerical key ("600") is still a numerical key, so it is reset to 0.

Reviewer: ```suggestion
In the code snippet below `array_shift()` is used twice. The first time it pops the one element with a numerical key, the second time an element with a literal key. After that the `$num_array` is printed out. Notice that the string `"600"` is still a numerical key, so it is reset to `0`.
```
